earth2mip/lagged_ensembles/score.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+import torch
+
+import earth2mip.grid
+from earth2mip.crps import crps_from_empirical_cdf
+
+logger = logging.getLogger(__name__)
+
+
+def weighted_average(x, w, dim):
+    return torch.mean(x * w, dim) / torch.mean(w, dim)
```

Comment:
Why does this function exist here? Should this not exist elsewhere? Modulus has weighted average utilities.

Reply:
I didn't know that modulus had weighted averaged utilities. It's one line of code, and only used as a helper here. Not public API.

Reply:
Since it is such a small line of code, intended to be private, then why does it need to exist as its own function at all? Again, if it is intended to be private then why is that not apparent? 

Can we just drop in `modulus.metrics.climate.reduction.global_mean`?

Reply:
It's called a couple of times here, and I didn't want to use copy-paste. Not advertising that it be used everywhere.

I like to see the cos_lat directly in the code, since e.g. graphcast does something different (sin(lat + dlat /2) - sin(lat-dlat/2).

Reply:
I still have reservations but I agree that this is low enough priority that we can resolve this without further note.